torch/distributed/_sharded_tensor/api.py

```diff
@@ -751,3 +928,122 @@ def _create_tensor_from_params(*size, local_device, tensor_init_params: TensorIn
                           device=local_device, )
     else:
         raise ValueError(f'Unsupported create_op: {tensor_init_params.create_op}')
+
+
+class _PartialTensor(object):
+    """
+    PartialTensor is an abstraction to represent Tensors that need
+    aggregation across multiple devices and multiple processes.
+
+    PartialTensor is initialized in an SPMD like fashion where each rank
+    initializes the PartialTensor. The PartialTensor object on each rank
+    then only stores the local partial shard, process group and the
+    aggregation way to get a full tensor.
+
+    PartialTensor doesn't provide any Tensor like operations but is a
+    wrapper providing the Tensor representing the local partial shard.
+
+    We assume the size of each local tensor to be exactly the same.
+
+    Users can apply custom distributed sharded computations on top of
+    this primitive.
+
+    Args:
+        local_partial_shard (Tensor): Partial result stored across ranks.
+        pg (ProcessGroup): The process group to aggregate on.
+        aggr_op (distributed_c10d.ReduceOp): Way to aggregate the partial result.
```

Comment:
nit: `all_reduce` API call this `op`. If you prefer to be more precise, `reduce_op` might be better, as it would match the `ReduceOp` type name.

Reply:
OK, thanks for your suggestion. Let me change it.